guarddog/analyzer/analyzer.py

```diff
@@ -45,7 +39,7 @@ def __init__(self, ecosystem=ECOSYSTEM.PYPI) -> None:
         self.metadata_detectors = get_metadata_detectors(ecosystem)
 
         self.metadata_ruleset = self.metadata_detectors.keys()
-        self.sourcecode_ruleset = SEMGREP_RULE_NAMES
+        self.sourcecode_ruleset = set(map(lambda r: r["id"], SOURCECODE_RULES[ecosystem]))
```

Comment:
how about `[rule["id"] for rule in SOURCECODE_RULES[ecosystem]]`?

Reply:
you're right, I did not notice it was expected to be a list in here.

Reply:
fixed in 7a31e9dedff5e00963ece1784ede0a24ae83f190